backend/src/main/java/com/benine/backend/http/EditPresetHandler.java

```diff
@@ -0,0 +1,101 @@
+package com.benine.backend.http;
+
+import com.benine.backend.LogEvent;
+import com.benine.backend.Preset;
+import com.benine.backend.ServerController;
+import com.benine.backend.camera.CameraConnectionException;
+import com.benine.backend.camera.ipcameracontrol.IPCamera;
+import com.benine.backend.video.StreamNotAvailableException;
+
+import org.eclipse.jetty.server.Request;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class EditPresetHandler extends RequestHandler {
+  
+  /**
+   * Constructor for a new editPresetHandler that handles editing a preset.
+   */
+  public EditPresetHandler() {}
+    
+  private ServerController control = ServerController.getInstance();
+
+  @Override
+  public void handle(String s, Request request, HttpServletRequest req, HttpServletResponse res)
+          throws IOException, ServletException {
+    
+    try {
+      String overwriteTag = request.getParameter("overwriteTag");
+      String overwritePosition = request.getParameter("overwriteposition");
+      String camID = request.getParameter("camera");
+      int presetID = Integer.parseInt(request.getParameter("presetid"));
+      String tags = request.getParameter("tags");
+      
+      Preset preset = control.getPresetController().getPresetById(presetID);
+     
+      List<String> tagList = new ArrayList<String>();
+      if (tags != null) {
+        tagList = Arrays.asList(tags.split("\\s*,\\s*")); 
+      }
+      if (overwriteTag.equals("true")) {
+        updateTag(preset, tagList);
+      }
+      
+      if (overwritePosition.equals("true")) {
+        updatePosition(camID,tagList,presetID);
+      }
+    } catch (MalformedURIException | SQLException | StreamNotAvailableException e) {
+      getLogger().log(e.getMessage(), LogEvent.Type.WARNING);
```

Comment:
Plrease use the log(String des, Exception e) method here to keep things consistent.